src/formatDate.js

```diff
@@ -50,7 +50,32 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const newDate = date.split(oldSeparator);
+  const oldDateFormat = fromFormat.slice(0, -1);
+  const newDateFormat = toFormat.slice(0, -1);
+
+  if (fromFormat[0][0] !== toFormat[0][0]) {
+    newDate.reverse();
+    oldDateFormat.reverse();
+  }
+
+  for (let i = 0; i < newDateFormat.length; i++) {
+    if (oldDateFormat[i].length < newDateFormat[i].length) {
+      if (newDate[i] < 30) {
+        newDate[i] = '20' + newDate[i];
+      } else {
+        newDate[i] = '19' + newDate[i];
+      }
+    }
+
+    if (oldDateFormat[i].length > newDateFormat[i].length) {
```

Comment:
How can someone guess which part of the day you're converting?👀 Dog years to human's?

Reply:
Also your program will hit this line of code even if the first `if` was true, not a good idea😁

Reply:
Here I mean make it somehow obvious that years might be converted, and only years, not days, months etc.

Reply:
Agree. I didn't notice at first that we must convert only years...I was preparing for more